UndertaleModTool/Editors/UndertaleEmbeddedTextureEditor.xaml.cs

```diff
@@ -154,29 +174,41 @@ private void Import_Click(object sender, RoutedEventArgs e)
             {
                 try
                 {
-                    Bitmap bmp;
-                    using (var ms = new MemoryStream(TextureWorker.ReadTextureBlob(dlg.FileName)))
+                    GMImage image;
+                    if (System.IO.Path.GetExtension(dlg.FileName).Equals(".png", StringComparison.InvariantCultureIgnoreCase))
                     {
-                        bmp = new Bitmap(ms);
+                        // Import PNG data verbatim, without attempting to modify it
+                        image = GMImage.FromPng(File.ReadAllBytes(dlg.FileName), true).ConvertToFormat(target.TextureData.Image.Format);
+                    }
+                    else
+                    {
+                        // Import any file type
+                        using var magickImage = new MagickImage(dlg.FileName);
+                        magickImage.Format = MagickFormat.Rgba;
+                        magickImage.Alpha(AlphaOption.Set);
+                        magickImage.SetCompression(CompressionMethod.NoCompression);
+
+                        // Import image
+                        image = GMImage.FromMagickImage(magickImage).ConvertToFormat(target.TextureData.Image.Format);
```

Comment:
```suggestion
                        image = GMImage.FromMagickImage(magickImage)
									   .ConvertToFormat(target.TextureData.Image.Format);
```

![image](https://github.com/user-attachments/assets/2013c75c-6405-481d-859c-cfd60c576c95)